app/models/katello/product.rb

```diff
@@ -212,13 +213,21 @@ def syncable_content?
       repositories.any?(&:url?)
     end
 
+    def product_content_by_id(content_id)
+      product_contents.find { |pc| pc.content.cp_content_id == content_id }
+    end
+
     def available_content(content_view_version_id = nil)
-      self.productContent.select do |content|
-        repos = self.repositories.subscribable.where(content_id: content.content.id)
+      product_contents.select do |product_content|
+        repos = self.repositories.subscribable.where(content_id: product_content.content.cp_content_id)
         repos.exists? && (content_view_version_id.nil? || repos.where(content_view_version_id: content_view_version_id.to_i).count > 0)
       end
     end
 
+    def displayable_product_contents
+      product_contents.select { |pc| pc.displayable? }
```

Comment:
can we make displayable? a scope,  so that we aren't looping over all the contents, we can just query based on type?